src/types.ts

```diff
@@ -0,0 +1,32 @@
+export type Cat = Readonly<{
+  id: string;
+  name: string;
+  url: string | null;
+}>;
+
+export type CatList = ReadonlyArray<Cat>;
+
+export type PageMetadata = Readonly<{
+  hasNextPage: boolean;
+}>;
+
+export type PagedResult = Readonly<{
+  results: CatList;
+  metadata: PageMetadata;
+}>;
```

Comment:
[suggestion] Instead of using fixed types, you can use generics to indicate dynamically what the `results` will look like. Putting `CatList` here, will make it not re-usable:
```suggestion
export type PagedResult<T> = Readonly<{
  results: T;
  metadata: PageMetadata;
}>;
```
And then in your service:
```typescript
const getAllCats = async ({
  pageNumber,
  pageSize,
  sortBy,
}: {
  pageNumber: number;
  pageSize: number;
  sortBy: string;
}): Promise<PagedResult<CatList>> => {
```
WDYT?

Reply:
done